File: EquationArranger/src/main/java/Quadratic/Equation/Scanner/ScanState.java
package Quadratic.Equation.Scanner;

public enum ScanState
{
    DuringStart,
    DuringNumber,
    DuringVariable,
    DuringCoefficientSign,
    DuringExponentSign,
    DuringExponentNumber,
    DuringEqualSign,
    DuringMultiplySign,
}

File: src/EquationArranger/scan_state.go
package equationscanner

type ScanState int

const (
    DuringStart =      	iota
    DuringNumber
    DuringVariable 
    DuringCoefficientSign
    DuringExponentSign
    DuringExponentNumber
    DuringEqualSign
    DuringMultiplySign
)